driver-async/src/test/functional/com/mongodb/async/client/MongoClientsSpecification.groovy

```diff
@@ -118,4 +126,71 @@ class MongoClientsSpecification extends FunctionalSpecification {
         cleanup:
         client?.close()
     }
+
+    @Unroll
+    def 'should correct parse read preference for #readPreference'() {
```

Comment:
It's not really about parsing.   To be consistent with other test method names, how about:

    def 'should apply read preference from connection string to settings'